### PR TITLE
Cache Query Database for Mentions-Generator (every 10 minutes refresh)

### PR DESCRIPTION
**Pull request by Max-Leopold**

### CODE REVIEW

File: mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/MentionsGeneratorApplication.java [hunk 1]
```diff
@@ -21,5 +31,19 @@ public static void main(String[] args) {
     public void sendMessagesOnKafka() {
         producer.send("Waz up?");
     }
+
+    @Bean
+    public CacheManager cacheManager() {
+        ConcurrentMapCacheManager cacheManager = new ConcurrentMapCacheManager("{queryID, query}");
+
+        return cacheManager;
```
Comment (YNedderhoff) [inline mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/MentionsGeneratorApplication.java]: why not just `return new ConcurrentMapCacheManager("{queryID, query}");``

File: mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/MentionsGeneratorApplication.java [hunk 2]
```diff
@@ -21,5 +31,19 @@ public static void main(String[] args) {
...
+    }
+
+    //Refreh Cache every 10 minutes
+    @CacheEvict(allEntries = true, cacheNames = "{queryID, query}")
+    @Scheduled(fixedDelay = 10 * 60 * 1000)
+    public void reportCacheEvict() {
+        System.out.println("Flush Cache " + Instant.now());
```
Comment (YNedderhoff) [inline mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/MentionsGeneratorApplication.java]: please use a logger instead

File: mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/postgres/cache/QueryCache.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class QueryCache {
+
+    @Autowired
+    QueryRepository queryRepository;
```
Comment (YNedderhoff) [inline mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/postgres/cache/QueryCache.java]: private

**Comment by Max-Leopold** [comment]
All Done

File: mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/MentionsGeneratorApplication.java [hunk 4]
```diff
@@ -1,13 +1,49 @@
...
+@EnableScheduling
 public class MentionsGeneratorApplication {
 
+    private static final Logger logger = LoggerFactory.getLogger(MentionsGeneratorApplication.class);
+
     public static void main(String[] args) {
         SpringApplication.run(MentionsGeneratorApplication.class, args);
     }
+
+    private Producer producer;
```
Comment (YNedderhoff) [inline mentions-generator/src/main/java/de/brandwatch/minianalytics/mentiongenerator/MentionsGeneratorApplication.java]: can be final

**Pull request merged**